linera-views-derive/src/lib.rs

```diff
@@ -702,234 +762,35 @@ pub mod tests {
-                #[async_trait::async_trait]
-                impl #generics linera_views::views::View<#context> for TestView #generics
-                #constraints
-                {
-                    fn context(&self) -> &#context {
-                        use linera_views::views::View;
-                        self.register.context()
-                    }
-                    async fn load(
-                        context: #context
-                    ) -> Result<Self, linera_views::views::ViewError> {
-                        use linera_views::{futures::join, common::Context};
-                        linera_views::increment_counter(
-                            &linera_views::LOAD_VIEW_COUNTER,
-                            stringify!(TestView),
-                            &context.base_key(),
-                        );
-                        let index = 0;
-                        let base_key = context.derive_tag_key(linera_views::common::MIN_VIEW_TAG, &index)?;
-                        let register_fut =
-                            RegisterView::load(context.clone_with_base_key(base_key));
-                        let index = 1;
-                        let base_key = context.derive_tag_key(linera_views::common::MIN_VIEW_TAG, &index)?;
-                        let collection_fut =
-                            CollectionView::load(context.clone_with_base_key(base_key));
-                        let result = join!(register_fut, collection_fut);
-                        let register = result.0?;
-                        let collection = result.1?;
-                        Ok(Self {
-                            register,
-                            collection
-                        })
-                    }
-                    fn rollback(&mut self) {
-                        self.register.rollback();
-                        self.collection.rollback();
-                    }
-                    fn flush(
-                        &mut self,
-                        batch: &mut linera_views::batch::Batch
-                    ) -> Result<(), linera_views::views::ViewError> {
-                        use linera_views::views::View;
-                        self.register.flush(batch)?;
-                        self.collection.flush(batch)?;
-                        Ok(())
-                    }
-                    fn delete(self, batch: &mut linera_views::batch::Batch) {
-                        use linera_views::views::View;
-                        self.register.delete(batch);
-                        self.collection.delete(batch);
-                    }
-                    fn clear(&mut self) {
-                        self.register.clear();
-                        self.collection.clear();
-                    }
-                }
-            );
-
-            assert_eq!(output.to_string(), expected.to_string());
+            insta::assert_display_snapshot!(generate_view_code(input, true));
```

Comment:
Would it make sense to pretty-print these? (E.g. using `prettyplease`?) Then the snapshots would be much more readable.

Reply:
Yes we did exactly that — this PR is atop #1237 where I introduced `prettyplease` here :)  I just haven't rebased it yet because I'm trying to figure out why it's not passing CI.